bond_app/authentication.py

```diff
@@ -74,17 +77,21 @@ def require_user_info(self, request_state, token_info_fn=_token_info):
 
         token = auth_header_parts[1]
 
-        user_info = self.cache_api.get(key='access_token:' + token)
-        if user_info is None:
-            user_info = self._fetch_user_info(token, token_info_fn)
+        google_user_info = self.cache_api.get(key='access_token:' + token)
+        if google_user_info is None:
+            google_user_info = self._fetch_user_info(token, token_info_fn)
             # cache for 10 minutes or until token expires
-            expires_in = min([user_info.expires_in, self.config.max_token_life])
+            expires_in = min([google_user_info.expires_in, self.config.max_token_life])
             logging.debug("caching token %s for %s seconds", token, expires_in)
-            self.cache_api.add(key='access_token:' + token, value=user_info, expires_in=expires_in)
+            self.cache_api.add(key='access_token:' + token, value=google_user_info, expires_in=expires_in)
         else:
             logging.debug("auth token cache hit for token %s", token)
 
-        return user_info
+        sam_user_info = self.sam_api.user_info(google_user_info)
+        if sam_user_info is None or not sam_user_info[SamKeys.USER_ENABLED_KEY]:
```

Comment:
I was worried about this too. I tried to test it in the python console, but I'm not 100% sure my test was sufficient. This is what I did

> `json.loads("{\"userSubjectId\": \"1234\", \"enabled\": true}")`
> `{'userSubjectId': '1234', 'enabled': True}`

This makes me believe that `json.loads` will recognize it as a boolean so that we can treat it as such later on when accessing the dict, but I can try to do more testing to be sure

Reply:
Right, my worry would be around:

`foo = json.loads("{\"userSubjectId\": \"1234\", \"enabled\": \"false\"}")`

because I think the truthiness rules would evaluate `foo.enabled` as `True` in this case.

Reply:
I did some extra testing and it does work like we want it too, so this should be fine, but good call on double checking it.

> result = requests.get(url=base_url + '/register/user/v2/self/info', headers=headers)
> sam_user_info = json.loads(result.content)
> sam_user_info
> {'enabled': True, 'userEmail': 'marc.testbott@gmail.com', 'userSubjectId': '106960538915967416130'}

and an if statement as a sanity check
> if sam_user_info["enabled"]:
...     print('user is enabled')
... 
user is enabled